### ISSUE TITLE
detect.py does not generate output

### ISSUE THREAD

**Issue by hyl-g** [ultralytics/yolov5#1088]
Hello,

I downloaded the latest docker version.  I can train the network with custom data but I can't get any thing detected using detect.py.  So I use the pre-trained model yolov5x.pt to detect the bus.jpg.  Nothing is detected even I set the confident to 0.01.

Below is my command:
python detect.py --img 640 --conf 0.01 --source inference/images/bus.jpg --weights yolov5x.pt

Appreciate if you can point out anything wrong here.

Thanks

**Comment by hyl-g** [comment]
Thanks for your comments.

I followed the links for the Jupyter Notebook and run the set up can command:

_!python detect.py --weights yolov5s.pt --img 416 --conf 0.4 --source inference/images/
Image(filename='inference/output/zidane.jpg', width=600)_

Here is the output:

_Namespace(agnostic_nms=False, augment=False, classes=None, conf_thres=0.4, device='', img_size=416, iou_thres=0.5, output='inference/output', save_txt=False, source='inference/images/', update=False, view_img=False, weights=['yolov5s.pt'])_
_Using CUDA device0 _CudaDeviceProperties(name='GeForce GTX 1660 SUPER', total_memory=5941MB)_

_Fusing layers... 
Model Summary: 191 layers, 7.46816e+06 parameters, 0 gradients
image 1/2 /usr/src/app/inference/images/bus.jpg: 416x352 Done. (0.024s)
image 2/2 /usr/src/app/inference/images/zidane.jpg: 288x416 Done. (0.024s)
Results saved to inference/output
Done. (1.194s)_

It doesn't detect buss, person or ties

**Comment by foochuanyue** [comment]
What custom data did you use? How many images? Epoch? and can you share your results.png in the run folder. That should give enough information for people to help you troubleshoot.

**Comment by glenn-jocher** [comment]
@hyl-g all operations are working correctly. Recommend you start with the Colab notebook, you can get this result in just two clicks:
https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb

<img width="811" alt="Screen Shot 2020-10-07 at 11 25 40 AM" src="https://user-images.githubusercontent.com/26833433/95313020-eb8a7f80-088f-11eb-9dae-14a33caaa062.png">

**Comment by hyl-g** [comment]
Thanks guys for your inputs.  After battling the problem for days without any luck with this docker version of detect.py, I decided to switch to clone the git hub version.  The detect.py runs and detects the objects without problem.  So I will give up the docker version for now.

Thanks

**Comment by glenn-jocher** [comment]
@hyl-g docker operates correctly, tested today. **All environments are operating correctly.**

<img width="1012" alt="Screen Shot 2020-10-08 at 11 35 53 AM" src="https://user-images.githubusercontent.com/26833433/95441494-7df75580-095a-11eb-88e8-c039949e3eb2.png">

**Comment by dongjuns** [comment]
@hyl-g That's too bad, I think sometimes docker could be an obstacle without familiar.
Anyway, in my docker case, It works.
```
python detect.py --source ./inference/test1/India/images/ --weights runs/exp21/weights/best.pt --img-size 704 --conf-thres 0.4 --save-txt
```

Good luck-! :smile:

**Comment by hyl-g** [comment]
Thanks guys,

I found the problem is associated with the use of GPU.  I tried the following two ways to run the docker container:
1.  sudo docker run --ipc=host -it ultralytics/yolov5:latest   <-- detect.py recognize the objects in the container
2. sudo docker run --ipc=host --gpus all -it ultralytics/yolov5:latest   <-- detect.py can't recognize the object
